app/src/main/java/io/homeassistant/companion/android/widgets/SingleItemArrayAdapter.kt

```diff
@@ -4,12 +4,81 @@ import android.content.Context
 import android.view.View
 import android.view.ViewGroup
 import android.widget.ArrayAdapter
+import android.widget.Filter
 import android.widget.TextView
+import java.util.Comparator
 
 class SingleItemArrayAdapter<T>(
     context: Context,
     private val createText: (T?) -> String
 ) : ArrayAdapter<T>(context, android.R.layout.simple_list_item_1) {
+
+    private var filterItems = ArrayList<T>()
+
+    private var containsFilter: Filter = object : Filter() {
```

Comment:
You could just inline this on line 78 `getFilter`

Reply:
Changed in commit b3d9d06